controllers/base_controller.js

```diff
@@ -652,10 +593,12 @@ module.exports = function BaseControllerModule(pb) {
      * @param {String} location
      * @param {Function} cb
      */
-    BaseController.prototype.redirect = function(location, cb){
-        cb(pb.RequestHandler.generateRedirect(location));
+    BaseController.prototype.redirect = function (location, cb) {
+        cb(pb.Errors.redirect(location));
+    };
+    BaseController.prototype.doRedirect = function (destination, code = 302) {
+        throw pb.Errors.redirect(destination, code);
```

Comment:
If I understand correctly you are returning and throwing an error here so nobody should use these methods anyich should ore , why no add a comment or meta data to show that the methods should not be used any more and which should be the proper way to perform the redirection

Reply:
So we have a minor issue in the way of KOA and its handling of redirection as integrated with Pencilblue.  The only way to get a redirect to happen is to force the middleware stack to end.  The only way to force the middleware stack to end is to throw an error (or not call next, but that is a bit more complicated given the safe guards in the system around isolating plugin code from system code.) 

So this was the approach that was taken.  Some more work can be done around handling and cleaning this up.  like setting a - flag on the body that will stop execution of the middleware stack.